Userland/Services/WebContent/InputSource.h

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include "AK/Debug.h"
+#include <AK/HashTable.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/String.h>
+
+namespace Web::WebDriver {
+
+// InputState has lists of InputSources, but the pointer input source constructor
+// depends on InputState, so the definitions are circular
+// forward declare InputState here
+class InputState;
+
+// https : // w3c.github.io/webdriver/#input-sources
+enum class InputSourceType { none,
+    key,
+    pointer,
+    wheel };
+// this enum is for the types of devices, which make up the subtype field of the PointerInputSource class
+enum class PointingDeviceSubtype { mouse,
+    pen,
+    touch };
+
+class InputSource {
+    // Each input source has an input id which is stored as a key in the input state map.
+    // input id is a UUID
+    AK::String input_id;
+
+    // create an input source given input state, type and optional subtype:
+    // FIXME: It probably makes more sense to designate this switch to factory
+    // that the Input State can call into
+    InputSource(AK::NonnullOwnPtr<InputState> /*input_state*/, InputSourceType /*type*/,
```

Comment:
InputState looks like it gets used multiple by multiple InputSources:
> An input state represents the overall state of a collection of [input sources](https://w3c.github.io/webdriver/#dfn-input-source).

Therefore, you don't want to pass it in an OwnPtr - a NonnullRefPtr would be more correct if InputSource needs to hold onto it. If it's just used temporarily, a regular reference (`InputState&`) would be fine. See https://github.com/LadybirdBrowser/ladybird/blob/master/Documentation/SmartPointers.md for an explanation of our pointer types and when to use them.